s4/clarity/step.py

```diff
@@ -415,6 +416,16 @@ def _get_attach_to_key(self):
     def _get_iomaps_shared_result_file_type(self):
         return "ResultFile"
 
+    @lazy_property
```

Comment:
Was there a reason you didn't represent this as an subnode_element property?

Reply:
No,
I think I was just in a frame of mind where I felt that `instrument` was tertiary to the working of the `step` more so `step/details` endpoint. The unit test for this case uses `stepdetails` xml, which shows `instrument` is clearly amenable to `subnode_link` or `subnode_element`. What do you recommend?
For reasons, less pertinent, to your question, the `# ToDo: This sould be cached so that we return the same object each time otherwise root.sub_element != root.sub_element` comment in `subnode_element` draws attention.

Reply:
You are correct that this should subnode_link not subnode_element. It is not necessary to cache subnode_link because it is cached by the underlying factory. The code below doesn't match our standard practice which is to use from_linknode but I think just switching to subnode_link is the correct behaviour.

Reply:
Interestingly the link in this case has name as it's value but I don't think we have a nice way to incorporate that into the returned object.